lib/order.rb

```diff
@@ -0,0 +1,83 @@
+# Jacquelyn Cheng - Nodes
+
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    if ![:pending, :paid, :processing, :shipped, :complete].include? fulfillment_status
+      raise ArgumentError, "Invalid fulfillment status"
+    end
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+  end
+
+  # returns an array of all orders
+  def self.all
+    all_orders = CSV.open("data/orders.csv").map do |row|
+      id = row[0].to_i
+      product_list = row[1].split(';')
+      products = {}
+      product_list.each do |product_data|
+        product_and_price = product_data.split(':')
+        products[product_and_price[0]] = product_and_price[1].to_f
+      end
+      customer = Customer.find(row[2].to_i)
+      fulfillment_status = row[3].to_sym
+      Order.new(id, products, customer, fulfillment_status)
+    end
+    return all_orders
+  end
+
+  # returns an Order instance matching the designated id
+  def self.find(order_id)
+    Order.all.each do |order|
```

Comment:
It is not find's job to call all, at least in my opinion. If you've already loaded the data, `self.find` should just find the person. If `@@orders` isn't loaded, I think it should return `nil`. This would be better separation of concerns.